frontend/middleware_logging.go

```diff
@@ -51,7 +50,12 @@ func MiddlewareLogging(w http.ResponseWriter, r *http.Request, next http.Handler
 
 	startTime := time.Now()
 
-	logger := r.Context().Value(ContextKeyLogger).(*slog.Logger).With(
+	loggerFromContext, ok := LoggerFromContext(r.Context())
+	if !ok {
+		loggerFromContext = DefaultLogger()
```

Comment:
Is there a valid scenario in which we will not have logger in the context?

Reply:
I would say that in theory no, because in _NewFrontend_ (frontend/frontend.go) we instantiate a context with a logger, but I think checking for that is better than panicking (in the worst case).